src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
+    await updateNetWorth(roomID, userID);
+    const userData = await getUserData(roomID, userID);
+    return userData.net_worth;
 }
 
-export const changeCash = async (roomID, userID, dayIndex, changeArray, prices) => {
-  const userRef = getUserRef(roomID, userID);
-  const userData = await getUserData(roomID,userID);
-  var totalMoney = userData.money_left;
-  prices.map(function(num,idx) {totalMoney -= num * changeArray[idx];});
-  userRef.update({money_left: totalMoney});
+export const getNumDays = async (roomID) => {
+    const roomData = await getRoomData(roomID);
+    return roomData.dates.length;
 }
 
-export const changeShares = async (roomID, userID, dayIndex, changeArray) => {
-  const userRef = getUserRef(roomID, userID);
-  const investRef = userRef.collection('investments').doc(dayIndex.toString());
-  const userData = await getUserData(roomID,userID);
-
-  // update currentShares array
-  var curArray = userData.currentShares;
-  var sum = curArray.map(function(num,idx) {return num + changeArray[idx];});
-  userRef.update({currentShares: sum});
-
-  // TODO: current have a separate call for investment. Consider accessing it through userData to save time
-  investRef.get().then(function(investDoc) {
-      if (investDoc.exists) {
-          var investData = investDoc.data();
-          var curArray = investData.change;
-          var sum = curArray.map(function(num,idx) {
-              return num + changeArray[idx];
-          });
-          investRef.update({
-              change: sum,
-          });
-      }
-  });
+export const getNumSymbols = async (roomID) => {
+    const roomData = await getRoomData(roomID);
+    return roomData.symbols.length;
 }
 
-export const makeInvestment = async (roomID, userID, dayIndex, changeArray) => {
-  const prices = await getPrices(roomID, dayIndex);
+// returns current price for all symbols being tracked
+export const getPrices = async (roomID, dayIndex) => {
+    const roomData = await getRoomData(roomID);
+    const symbolNamesArray = roomData.symbols;
+
+    var prices = [];
 
-  if (!(await verifyOk(roomID,userID,dayIndex,changeArray,prices))) return false;
+    // save all current prices in an array
+    for (var index = 0; index < symbolNamesArray.length; index++) {
+        const currentName = symbolNamesArray[index];
+        const symbolPricesDoc = await db.collection('Rooms').doc(roomID).collection(currentName).doc('Prices').get();
+        const symbolPricesData = symbolPricesDoc.data();
+        const currentPrice = symbolPricesData.prices[dayIndex];
+        prices.push(currentPrice);
+    }
 
-  changeCash(roomID,userID,dayIndex,changeArray,prices);
-  changeShares(roomID,userID,dayIndex,changeArray);
-  updateNetWorth(roomID, userID, dayIndex, prices);
+    return prices;
+}
 
-  return true;
+export const getRoomData = async (roomID) => {
+    const roomRef = getRoomRef(roomID);
+    const roomDoc = await roomRef.get();
+    const roomData = await roomDoc.data();
```

Comment:
don't need to await this
```suggestion
    const roomData = roomDoc.data();
```